apidef/oas/middleware.go

```diff
@@ -108,16 +108,65 @@ type PluginConfig struct {
 	//
 	// Tyk native API definition: `custom_middleware.driver`.
 	Driver apidef.MiddlewareDriver `bson:"driver,omitempty" json:"driver,omitempty"`
+
+	// Bundle configures custom plugin bundles.
+	Bundle *PluginBundle `bson:"bundle,omitempty" json:"bundle,omitempty"`
 }
 
-// Fill fills pluginConfig from apidef.
+// Fill fills PluginConfig from apidef.
 func (p *PluginConfig) Fill(api apidef.APIDefinition) {
 	p.Driver = api.CustomMiddleware.Driver
+
+	if p.Bundle == nil {
+		p.Bundle = &PluginBundle{}
+	}
+
+	p.Bundle.Fill(api)
+	if ShouldOmit(p.Bundle) {
+		p.Bundle = nil
+	}
 }
 
 // ExtractTo extracts *PluginConfig into *apidef.
 func (p *PluginConfig) ExtractTo(api *apidef.APIDefinition) {
 	api.CustomMiddleware.Driver = p.Driver
+
+	if p.Bundle != nil {
+		p.Bundle.ExtractTo(api)
+	}
+}
+
+// PluginBundle holds configuration for custom plugins.
+type PluginBundle struct {
+	// Enabled enables the custom plugin bundles.
+	//
+	// Tyk classic API definition: `custom_middleware_bundle_disabled`
+	Enabled bool `bson:"enabled" json:"enabled"` // required.
+	// Path is the path suffix to construct the URL to fetch plugin bundle from.
+	// Path will be suffixed to `bundle_base_url` in gateway config.
+	Path string `bson:"path" json:"path"` // required.
+}
+
+// Fill fills PluginBundle from apidef.
+func (p *PluginBundle) Fill(api apidef.APIDefinition) {
+	if !api.CustomMiddlewareBundleDisabled && api.CustomMiddlewareBundle == "" {
```

Comment:
Shouldn't this be `api.CustomMiddlewareBundleDisabled? && api.CustomMiddlewareBundle == ""`?

Reply:
we're checking for the default values here, so `!api.CustomMiddlewareBundleDisabled && api.CustomMiddlewareBundle == "" ` is the expected condition right?